src/rgw/driver/d4n/d4n_policy.cc

```diff
@@ -473,176 +574,223 @@ void LFUDAPolicy::cleaning(const DoutPrefixProvider* dpp)
     ldpp_dout(dpp, 10) << __LINE__ << " " << __func__ << "(): e->obj_key=" << e->obj_key << dendl;
-    l.unlock();
-    if (!e->key.empty() && (e->dirty == true) && (std::difftime(time(NULL), e->creationTime) > interval)) { //if block is dirty and written more than interval seconds ago
-      rgw_user c_rgw_user = e->user;
-      //writing data to the backend
-      //we need to create an atomic_writer
-      std::unique_ptr<rgw::sal::User> c_user = driver->get_user(c_rgw_user);
-
-      std::unique_ptr<rgw::sal::Bucket> c_bucket;
-      rgw_bucket c_rgw_bucket = rgw_bucket(c_rgw_user.tenant, e->bucket_name, e->bucket_id);
-
-      RGWBucketInfo c_bucketinfo;
-      c_bucketinfo.bucket = c_rgw_bucket;
-      c_bucketinfo.owner = c_rgw_user;
-      int ret = driver->load_bucket(dpp, c_rgw_bucket, &c_bucket, null_yield);
-      if (ret < 0) {
-        ldpp_dout(dpp, 10) << __func__ << "(): load_bucket() returned ret=" << ret << dendl;
-        //Remove bucket should be implemented in d4n which will take care of deleting objects belonging to the bucket, and hence we should not reach here
-        erase_dirty_object(dpp, e->key, null_yield);
+
+    if (!e->key.empty()) {
+      auto p = o_entries_map.find(e->key);
+      if (p == o_entries_map.end()) {
         continue;
       }
 
-      std::unique_ptr<rgw::sal::Object> c_obj = c_bucket->get_object(e->obj_key);
-      ldpp_dout(dpp, 20) << __func__ << "(): c_obj oid =" << c_obj->get_oid() << dendl;
-
-      ACLOwner owner{c_user->get_id(), c_user->get_display_name()};
-
-      std::unique_ptr<rgw::sal::Writer> processor =  driver->get_atomic_writer(dpp,
-              null_yield,
-              c_obj.get(),
-              owner,
-              NULL,
-              0,
-              "");
-
-      int op_ret = processor->prepare(null_yield);
-      if (op_ret < 0) {
-	ldpp_dout(dpp, 20) << __func__ << "processor->prepare() returned ret=" << op_ret << dendl;
-        erase_dirty_object(dpp, e->key, null_yield);
+      if (p->second.second == State::INVALID) {
+        invalid = true;
       }
+      l.unlock();
+    
+      if ((e->dirty == true)) { //if block is dirty and written more than interval seconds ago
+	rgw_user c_rgw_user = e->user;
+	//writing data to the backend
+	//we need to create an atomic_writer
+	std::unique_ptr<rgw::sal::User> c_user = driver->get_user(c_rgw_user);
+
+	std::unique_ptr<rgw::sal::Bucket> c_bucket;
+	rgw_bucket c_rgw_bucket = rgw_bucket(c_rgw_user.tenant, e->bucket_name, e->bucket_id);
+
+	RGWBucketInfo c_bucketinfo;
+	c_bucketinfo.bucket = c_rgw_bucket;
+	c_bucketinfo.owner = c_rgw_user;
+	int ret = driver->load_bucket(dpp, c_rgw_bucket, &c_bucket, null_yield);
+	if (ret < 0) {
+	  ldpp_dout(dpp, 10) << __func__ << "(): load_bucket() returned ret=" << ret << dendl;
+	  //Remove bucket should be implemented in d4n which will take care of deleting objects belonging to the bucket, and hence we should not reach here
```

Comment:
As you can see that loading of a bucket throwing an error must not happen when remove bucket is is place, so please restore the original code path.